articles/055_ros_parameter_design.md

```diff
@@ -0,0 +1,202 @@
+
+This functionality will be exposed through local API calls as well as a ROS Service call API for remote operations.
+
+### Parameter update validation
+
+The node will allow for the registration of a callback for custom parameter value validation.
+
+### Backwards compatibility Parameter Server like behavior
+
+
+There are use cases where the older behavior with parameter server was useful.
+Both persisting beyond the duration of a specific node is valuable as well as having parameters with no specific association to a node which would potentially own or validate the values.
+To this end we propose to write a simple node which emulates the policy of the ROS 1.0 parameter server: it runs in namespace `/` and simply accepts all changes requested.
+
+### Parameter API
+
+The client libraries will provide an API for interfacing with the Core Parameter API for both local and remote nodes including return codes.
+
+### Parameter Events
+
+Each node will provide a topic on which parameter events will be published.
+This topic is to support monitoring parameters for change.
+It is expected that client libraries will implement the ability to register callbacks for specific parameter changes using this topic.
+
+### Logging and playback
+
+When logging an entire system, the parameter changes can be logged via standard topic recording of the events channel.
+An implementation of the playback mechanism could listen to the parameter event streams and invoke the set parameter calls via the remote API.
+
+## Current implementation
+
+The above specification has been prototyped; the implementation can be found in [rclcpp](https://github.com/ros2/rclcpp).
+The defintion of the services to use for interacting remotely are contained in the [rcl_interfaces package](https://github.com/ros2/rcl_interfaces)
+
+### Unimplemented
+
+Currently there a few parts of the specification unimplemented.
+
+- No parameter subscription registration.
+  The events are published, but there is not way to register a callback for changes to a specific parameter.
+  You can currently register for a callback on all changes for parameters of a node.
+- The ability to register callback to validate parameter updates prior to them being updated is not available.
+- There has been no work on logging and playback of logged parameter changes.
+- The ability to list and get expected validation policy has not been implemented.
+  It is expected to operate at a slightly higher level than parameters, and it possibly will be related to the component life cycle.
+
+## Topics not covered at the moment
+
+Going forward, there are still topics to discuss and flesh out either in this document or others. A few to highlight.
+
+### Parameter initialization
+
+There are several ways to load parameters at startup including command line arguments, roslaunch arguments, and potentially parameter files.
+This is something which should be addressed in conjunction with the new launch system.
+
```

Comment:
the more 'required' bits of metadata at initialization the better, as it helps with self documentation. i recommend that initialization occur at a single location in the node.